File: CS/09_Forms/AddJavaScriptAction.cs
using System;
using System.Drawing;
using System.IO;
using System.Windows.Forms;
using System.Xml.XPath;
using Spire.Pdf;
using Spire.Pdf.Actions;
using Spire.Pdf.AutomaticFields;
using Spire.Pdf.Fields;
using Spire.Pdf.Graphics;

namespace AddJavaScriptAction
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Specify the input file path
            string input = @"..\..\..\..\..\..\Data\FormFieldTemplate.pdf";

            // Create a new PdfDocument object
            PdfDocument pdf = new PdfDocument();

            // Load the PDF document from the specified input file
            pdf.LoadFromFile(input);

            // Get the first page of the PDF document
            PdfPageBase page = pdf.Pages[0];

            // Allow creation of form fields in the document
            pdf.AllowCreateForm = true;

            // Create a font object with Helvetica family, font size 12, and bold style
            PdfFont font = new PdfFont(PdfFontFamily.Helvetica, 12f, PdfFontStyle.Bold);

            // Create a brush object for drawing black color
            PdfBrush brush = PdfBrushes.Black;

            // Specify the starting coordinates for drawing text on the page
            float x = 50;
            float y = 550;
            float tempX = 0;

            // Draw a text string on the page
            string text1 = "Enter a number, such as 12345: ";
            page.Canvas.DrawString(text1, font, brush, x, y);

            // Add a textBox field to the page
            tempX = font.MeasureString(text1).Width + x + 15;
            PdfTextBoxField textbox = new PdfTextBoxField(page, "Number-TextBox");
            textbox.Bounds = new RectangleF(tempX, y, 100, 15);
            textbox.BorderWidth = 0.75f;
            textbox.BorderStyle = PdfBorderStyle.Solid;

            // Set a JavaScript action for handling key press events in the text field
            string js = PdfJavaScript.GetNumberKeystrokeString(2, 0, 0, 0, "$", true);
            PdfJavaScriptAction jsAction = new PdfJavaScriptAction(js);
            textbox.Actions.KeyPressed = jsAction;

            // Add a JavaScript action to format the value of the text field
            js = PdfJavaScript.GetNumberFormatString(2, 0, 0, 0, "$", true);
            jsAction = new PdfJavaScriptAction(js);
            textbox.Actions.Format = jsAction;

            // Add the text box field to the form fields collection of the PDF document
            pdf.Form.Fields.Add(textbox);

            // Specify the output file path for saving the modified PDF document
            string output = "AddJavaScriptAction_out.pdf";

            // Save the PDF document to the specified output file
            pdf.SaveToFile(output);


            //Launch the Pdf file
            PDFDocumentViewer(output);
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }

        }
    }
}

File: VB/09_Forms/AddJavaScriptAction.vb
Imports Spire.Pdf
Imports Spire.Pdf.Actions
Imports Spire.Pdf.Fields
Imports Spire.Pdf.Graphics

Namespace AddJavaScriptAction
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub
        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Specify the input file path
            Dim input As String = "..\..\..\..\..\..\Data\FormFieldTemplate.pdf"

            ' Create a new PDF document
            Dim pdf As New PdfDocument()

            ' Load the PDF document from the input file
            pdf.LoadFromFile(input)

            ' Get the first page of the document
            Dim page As PdfPageBase = pdf.Pages(0)

            ' Allow form field creation in the document
            pdf.AllowCreateForm = True

            ' Set the font and brush for drawing text
            Dim font As New PdfFont(PdfFontFamily.Helvetica, 12.0F, PdfFontStyle.Bold)
            Dim brush As PdfBrush = PdfBrushes.Black

            ' Set the initial coordinates for drawing
            Dim x As Single = 50
            Dim y As Single = 550
            Dim tempX As Single = 0

            ' Define the text to be drawn on the page
            Dim text1 As String = "Enter a number, such as 12345: "

            ' Draw the text onto the page
            page.Canvas.DrawString(text1, font, brush, x, y)

            ' Calculate the X coordinate for the textbox based on the width of the text
            tempX = font.MeasureString(text1).Width + x + 15

            ' Create a textbox form field
            Dim textbox As New PdfTextBoxField(page, "Number-TextBox")
            textbox.Bounds = New RectangleF(tempX, y, 100, 15)
            textbox.BorderWidth = 0.75F
            textbox.BorderStyle = PdfBorderStyle.Solid

            ' Set the JavaScript keystroke action for the textbox
            Dim js As String = PdfJavaScript.GetNumberKeystrokeString(2, 0, 0, 0, "$", True)
            Dim jsAction As New PdfJavaScriptAction(js)
            textbox.Actions.KeyPressed = jsAction

            ' Set the JavaScript format action for the textbox
            js = PdfJavaScript.GetNumberFormatString(2, 0, 0, 0, "$", True)
            jsAction = New PdfJavaScriptAction(js)
            textbox.Actions.Format = jsAction

            ' Add the textbox field to the PDF form
            pdf.Form.Fields.Add(textbox)

            ' Specify the output file path
            Dim output As String = "AddJavaScriptAction_out.pdf"

            ' Save the modified PDF document to the output file
            pdf.SaveToFile(output)

            ' Close the document
            pdf.Close()

            ' Launch the Pdf file
            PDFDocumentViewer(output)
        End Sub
        Private Sub PDFDocumentViewer(ByVal fileName As String)
            Try
                Process.Start(fileName)
            Catch
            End Try

        End Sub
    End Class
End Namespace
